core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -505,6 +507,40 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         return Service.START_NOT_STICKY;
     }
 
+    private void skipIntro(Playable playable) {
+        int skipIntro = 0;
+        if (! (playable instanceof FeedMedia)) {
+            return;
+        }
+
+        if (playable.getLastPlayedTime() > 0) {
+            return;
+        }
+
+        if (playable instanceof FeedMedia) {
+            FeedMedia feedMedia = (FeedMedia) playable;
+            FeedPreferences preferences = feedMedia.getItem().getFeed().getPreferences();
+            skipIntro = preferences.getFeedSkipIntro();
+        }
+
+        String skipIntroMesg = "";
+        Context context = getApplicationContext();
+        if (skipIntro > 0) {
+            int duration = getDuration();
+            if (skipIntro * 1000 < duration) {
+                mediaPlayer.seekTo(skipIntro * 1000);
+                skipIntroMesg = context.getString(R.string.pref_feed_skip_intro_toast,
+                        skipIntro,
+                        context.getString(R.string.time_seconds));
+            }
+        }
+        if (skipIntroMesg != "") {
```

Comment:
I would move this up into the two `if`s above. Then you can skip the check (`skipIntroMesg != ""` if and only if the code above is executed).